cmd/habitat/community/executor.go

```diff
@@ -1,54 +1,152 @@
 package community
 
 import (
-	"encoding/base64"
 	"encoding/json"
-	"fmt"
-	"path/filepath"
 
 	"github.com/eagraf/habitat/cmd/habitat/community/state"
+	"github.com/eagraf/habitat/cmd/habitat/node"
 	"github.com/eagraf/habitat/cmd/habitat/procs"
-	"github.com/eagraf/habitat/pkg/compass"
+	"github.com/eagraf/habitat/structs/community"
 	"github.com/rs/zerolog/log"
 )
 
+type TransitionExecutor func(update *state.StateUpdate) error
+
 type CommunityExecutor struct {
-	processManager *procs.Manager
+	node *node.Node
 }
 
-func NewCommunityExecutor(p *procs.Manager) *CommunityExecutor {
+func NewCommunityExecutor(n *node.Node) *CommunityExecutor {
 	return &CommunityExecutor{
-		processManager: p,
+		node: n,
+	}
+}
+
+// GetTransitionExecutor maps transition types to executor functions
+func (e *CommunityExecutor) GetTransitionExecutor(transitionType string) TransitionExecutor {
```

Comment:
wonder if we can get rid of this and just directly call `Execute`

Reply:
yeah this is silly

Reply:
this is probably gonna need some fancy reflection to get right, gonna punt on it